enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/IdentityVerificationLimitService.java

```diff
@@ -132,17 +127,64 @@ public void checkIdentityVerificationLimit(OwnerId ownerId) throws RemoteCommuni
             throw new IdentityVerificationLimitException("Max failed attempts reached for document upload");
         }
     }
 
+    /**
+     * Reset identity verification by setting activation flag to VERIFICATION_PENDING.
+     *
+     * @param ownerId Owner identification.
+     * @throws RemoteCommunicationException Thrown when communication with PowerAuth server fails.
+     * @throws IdentityVerificationException Thrown when identity verification reset fails.
+     * @throws OnboardingProcessLimitException Thrown when maximum failed attempts for identity verification have been reached.
+     * @throws OnboardingProcessException Thrown when onboarding process is invalid.
+     */
+    public void resetIdentityVerification(OwnerId ownerId) throws RemoteCommunicationException, IdentityVerificationException, OnboardingProcessLimitException, OnboardingProcessException {
+        Optional<OnboardingProcessEntity> processOptional = onboardingProcessRepository.findProcessByActivationId(ownerId.getActivationId());
+        if (processOptional.isEmpty()) {
+            logger.warn("Onboarding process not found, activation ID: {}", ownerId.getActivationId());
+            throw new OnboardingProcessException();
+        }
+        OnboardingProcessEntity process = processOptional.get();
+
+        // Increase process error score
+        processLimitService.incrementErrorScore(process, OnboardingProcessError.ERROR_IDENTITY_VERIFICATION_RESET);
+
+        // Check process error limits
+        process = processLimitService.checkOnboardingProcessErrorLimits(process);
+        if (process.getStatus() == OnboardingStatus.FAILED && OnboardingProcessEntity.ERROR_MAX_PROCESS_ERROR_SCORE_EXCEEDED.equals(process.getErrorDetail())) {
+            handleFailedProcess(ownerId);
+        }
+
+        // Check limits on identity verifications (error handling is handled by service)
+        checkIdentityVerificationLimit(ownerId);
+
+        // Update activation flags for reset of identity verification
+        activationFlagService.updateActivationFlagsForFailedIdentityVerification(ownerId);
+    }
+
+    /**
+     * Handle a process which just failed due to error score limit.
+     * @param ownerId Owner identification.
+     * @throws OnboardingProcessLimitException Exception thrown due to failed process.
+     * @throws RemoteCommunicationException Thrown when communication with PowerAuth server fails.
+     */
+    private void handleFailedProcess(OwnerId ownerId) throws OnboardingProcessLimitException, RemoteCommunicationException {
+        // Remove flag VERIFICATION_IN_PROGRESS and add VERIFICATION_PENDING flag
+        activationFlagService.updateActivationFlagsForFailedIdentityVerification(ownerId);
+
+        logger.warn("Max error score reached for onboarding process, {}.", ownerId);
+        throw new OnboardingProcessLimitException("Max error score reached for onboarding process");
```

Comment:
Shouldn't we add the process ID to the log message and/or exception message? Anyway, I would omit the logger, because is it logged at the controller level, isn't?

Reply:
I will update the message with more details and avoid logging twice.